src/polyform.jl

```diff
@@ -176,18 +175,18 @@ isexpr(x::PolyForm) = true
 iscall(x::Type{<:PolyForm}) = true
 iscall(x::PolyForm) = true
 
-function maketerm(::Type{<:PolyForm}, f, args, symtype, metadata)
-    basicsymbolic(t, f, args, symtype, metadata)
+function maketerm(t::Type{<:PolyForm}, f, args, metadata)
+    # TODO: this looks uncovered.
+    basicsymbolic(f, args, nothing, metadata)
 end
```

Comment:
This is uncovered by tests and thus had a wrong call to `basicsymbolic`. Can we remove this?

Reply:
I noticed this pull request addresses a few different things and has been open for a while now. To help get it merged more quickly, I think it would streamline the review process if we focused on one issue per PR.

You can always open new issues or pull requests for the other changes later. This will make it easier for us to:
- Review the changes effectively: Focusing on a single issue per PR allows for more thorough reviews and reduces the chance of overlooking potential issues.
- Merge changes faster: Smaller, focused PRs are generally quicker to review and merge.
- Maintain a cleaner commit history: Keeping commits focused makes it easier to understand the evolution of the codebase.

Reply:
@bowenszhu unfortunately this is the minimal changeset required to bump TI to 2.0 . i like smaller PRs as well